From 3d0c5a481308cde52699d9d87da01af878e6bcf1 Mon Sep 17 00:00:00 2001
From: Prarit Bhargava <prarit@redhat.com>
Date: Thu, 28 Aug 2014 14:22:38 -0400
Subject: [cpufreq] powernow: Use generic cpufreq routines

Message-id: <1409235879-1283-65-git-send-email-prarit@redhat.com>
Patchwork-id: 89476
O-Subject: [RHEL7.1 PATCH BZ 1134639 064/185] cpufreq: powernow: Use generic cpufreq routines
Bugzilla: 1134369
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Mark Langsdorf <mlangsdo@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1134639

commit d63bd27fe953daa402a108e141c36dcc59c6931c
Author: Viresh Kumar <viresh.kumar@linaro.org>
Date:   Thu Oct 3 20:28:17 2013 +0530

    cpufreq: powernow: Use generic cpufreq routines

    Most of the CPUFreq drivers do similar things in .exit() and .verify() routines
    and .attr. So its better if we have generic routines for them which can be used
    by cpufreq drivers then.

    This patch uses these generic routines in the powernow driver.

    Signed-off-by: Viresh Kumar <viresh.kumar@linaro.org>
    Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>

Cc: Lenny Szubowicz <lszubowi@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/cpufreq/powernow-k8.c b/drivers/cpufreq/powernow-k8.c
index c770b70..2efb759 100644
--- a/drivers/cpufreq/powernow-k8.c
+++ b/drivers/cpufreq/powernow-k8.c
@@ -1053,17 +1053,6 @@ static int powernowk8_target(struct cpufreq_policy *pol,
  return work_on_cpu(pol->cpu, powernowk8_target_fn, &pta);
 }
 
-/* Driver entry point to verify the policy and range of frequencies */
-static int powernowk8_verify(struct cpufreq_policy *pol)
-{
- struct powernow_k8_data *data = per_cpu(powernow_data, pol->cpu);
-
- if (!data)
-  return -EINVAL;
-
- return cpufreq_frequency_table_verify(pol, data->powernow_table);
-}
-
 struct init_on_cpu {
  struct powernow_k8_data *data;
  int rc;
@@ -1225,20 +1214,15 @@ out:
  return khz;
 }
 
-static struct freq_attr *powernow_k8_attr[] = {
- &cpufreq_freq_attr_scaling_available_freqs,
- NULL,
-};
-
 static struct cpufreq_driver cpufreq_amd64_driver = {
- .verify  = powernowk8_verify,
+ .verify  = cpufreq_generic_frequency_table_verify,
  .target  = powernowk8_target,
  .bios_limit = acpi_processor_get_bios_limit,
  .init  = powernowk8_cpu_init,
  .exit  = powernowk8_cpu_exit,
  .get  = powernowk8_get,
  .name  = "powernow-k8",
- .attr  = powernow_k8_attr,
+ .attr  = cpufreq_generic_attr,
 };
 
 static void __request_acpi_cpufreq(void)
-- 
1.7.1